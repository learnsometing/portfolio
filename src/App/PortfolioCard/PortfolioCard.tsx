import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import Img from 'gatsby-image';
import Grid from '@material-ui/core/Grid';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardHeader from '@material-ui/core/CardHeader';
import CardContent from '@material-ui/core/CardContent';
import ChildImageSharp from '../../shared/ChildImageSharp';

interface Image {
  src: ChildImageSharp;
  altText: string;
}

export interface ProjectCard {
  title: string;
  cardPhoto: Image;
  fields: {
    slug: string;
  };
}

interface CardProps {
  project: ProjectCard;
}

const PortfolioCard: React.FC<CardProps> = ({ project }) => {
  return (
    <Grid item component={'article'} xl={4} lg={4} md={6} sm={6} xs={12}>
      <Link to={project.fields.slug}>
        <Card raised>
          <CardActionArea>
            <Img
              fluid={project.cardPhoto.src.childImageSharp.fluid}
              alt="First slide"
            />
            <CardContent>
              <CardHeader title={project.title} />
            </CardContent>
          </CardActionArea>
        </Card>
      </Link>
    </Grid>
  );
};

PortfolioCard.propTypes = {
  project: PropTypes.shape({
    title: PropTypes.string.isRequired,
    cardPhoto: PropTypes.shape({
      src: PropTypes.shape({
        childImageSharp: PropTypes.shape({
          fluid: PropTypes.shape({
            base64: PropTypes.string.isRequired,
            aspectRatio: PropTypes.number.isRequired,
            src: PropTypes.string.isRequired,
            srcSet: PropTypes.string.isRequired,
            sizes: PropTypes.string.isRequired,
          }).isRequired,
        }).isRequired,
      }).isRequired,
      altText: PropTypes.string.isRequired,
    }).isRequired,
    fields: PropTypes.shape({
      slug: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default PortfolioCard;
