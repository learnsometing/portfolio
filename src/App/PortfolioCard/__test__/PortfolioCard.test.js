import React from 'react';
import PortfolioCard from '../PortfolioCard';
import TestRenderer from 'react-test-renderer';
import project from '../__factory__/PortfolioCard';

describe('PortfolioCard', () => {
  const proj = project();
  it('should match the snapshot', () => {
    const tree = TestRenderer.create(<PortfolioCard project={proj} />);
    expect(tree).toMatchSnapshot();
  });
});
