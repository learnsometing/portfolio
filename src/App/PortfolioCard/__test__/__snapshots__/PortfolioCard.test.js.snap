// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PortfolioCard should match the snapshot 1`] = `
<article
  className="col"
>
  <a
    className="PortfolioCard__ProjectLink-o19g6j-0 cwsxkS"
    href="/cash-pawn-jewelry-website/"
  >
    <div
      className="card"
    >
      <div
        className="PortfolioCard__CardHeader-o19g6j-1 ljCdif card-header"
      >
        Cash Pawn & Jewelry Website
      </div>
      <div
        className="PortfolioCard__MainPhoto-o19g6j-2 cKTzQN gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          aria-hidden={true}
          style={
            Object {
              "paddingBottom": "57.708333333333336%",
              "width": "100%",
            }
          }
        />
        <img
          alt=""
          aria-hidden={true}
          src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAMABQDASIAAhEBAxEB/8QAFwAAAwEAAAAAAAAAAAAAAAAAAAQFA//EABYBAQEBAAAAAAAAAAAAAAAAAAABAv/aAAwDAQACEAMQAAABfn1McHAI/8QAGhAAAQUBAAAAAAAAAAAAAAAAAAECAwQSEf/aAAgBAQABBQKw7MNaR6yD1NL0/8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPwE//8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAgEBPwE//8QAGBABAAMBAAAAAAAAAAAAAAAAAQACISD/2gAIAQEABj8CsmTbLz//xAAcEAEAAQQDAAAAAAAAAAAAAAABEQAQITFhcYH/2gAIAQEAAT8hZ1IjNMgSNNkEMQ8UAIg86t//2gAMAwEAAgADAAAAEDQf/8QAFBEBAAAAAAAAAAAAAAAAAAAAEP/aAAgBAwEBPxA//8QAFhEBAQEAAAAAAAAAAAAAAAAAARAh/9oACAECAQE/EByf/8QAHBABAAEEAwAAAAAAAAAAAAAAASEAEBExQWHR/9oACAEBAAE/EMUGYHZNMgzReybGoQSIeSjQZkIJt9Nv/9k="
          style={
            Object {
              "height": "100%",
              "left": 0,
              "objectFit": "cover",
              "objectPosition": "center",
              "opacity": 1,
              "position": "absolute",
              "top": 0,
              "transitionDelay": "500ms",
              "width": "100%",
            }
          }
        />
        <picture>
          <source
            sizes="(max-width: 1920px) 100vw, 1920px"
            srcSet="/static/56d0aba06cc4aae5197224cac0f73af4/9104c/cash-pawn_1.jpg 480w,
/static/56d0aba06cc4aae5197224cac0f73af4/a6352/cash-pawn_1.jpg 960w,
/static/56d0aba06cc4aae5197224cac0f73af4/d8255/cash-pawn_1.jpg 1920w,
/static/56d0aba06cc4aae5197224cac0f73af4/af56c/cash-pawn_1.jpg 2880w,
/static/56d0aba06cc4aae5197224cac0f73af4/4ee3c/cash-pawn_1.jpg 3554w"
          />
          <img
            alt="First slide"
            loading="lazy"
            onLoad={[Function]}
            sizes="(max-width: 1920px) 100vw, 1920px"
            src="/static/56d0aba06cc4aae5197224cac0f73af4/d8255/cash-pawn_1.jpg"
            srcSet="/static/56d0aba06cc4aae5197224cac0f73af4/9104c/cash-pawn_1.jpg 480w,
/static/56d0aba06cc4aae5197224cac0f73af4/a6352/cash-pawn_1.jpg 960w,
/static/56d0aba06cc4aae5197224cac0f73af4/d8255/cash-pawn_1.jpg 1920w,
/static/56d0aba06cc4aae5197224cac0f73af4/af56c/cash-pawn_1.jpg 2880w,
/static/56d0aba06cc4aae5197224cac0f73af4/4ee3c/cash-pawn_1.jpg 3554w"
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 0,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 500ms",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><source srcset=\\"/static/56d0aba06cc4aae5197224cac0f73af4/9104c/cash-pawn_1.jpg 480w,
/static/56d0aba06cc4aae5197224cac0f73af4/a6352/cash-pawn_1.jpg 960w,
/static/56d0aba06cc4aae5197224cac0f73af4/d8255/cash-pawn_1.jpg 1920w,
/static/56d0aba06cc4aae5197224cac0f73af4/af56c/cash-pawn_1.jpg 2880w,
/static/56d0aba06cc4aae5197224cac0f73af4/4ee3c/cash-pawn_1.jpg 3554w\\" sizes=\\"(max-width: 1920px) 100vw, 1920px\\" /><img loading=\\"lazy\\" sizes=\\"(max-width: 1920px) 100vw, 1920px\\" srcset=\\"/static/56d0aba06cc4aae5197224cac0f73af4/9104c/cash-pawn_1.jpg 480w,
/static/56d0aba06cc4aae5197224cac0f73af4/a6352/cash-pawn_1.jpg 960w,
/static/56d0aba06cc4aae5197224cac0f73af4/d8255/cash-pawn_1.jpg 1920w,
/static/56d0aba06cc4aae5197224cac0f73af4/af56c/cash-pawn_1.jpg 2880w,
/static/56d0aba06cc4aae5197224cac0f73af4/4ee3c/cash-pawn_1.jpg 3554w\\" src=\\"/static/56d0aba06cc4aae5197224cac0f73af4/d8255/cash-pawn_1.jpg\\" alt=\\"First slide\\" style=\\"position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
  </a>
</article>
`;
